### PR TITLE
Add SYSLIB* analyzers to global configs

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

**Comment by stephentoub** [comment]
@marek-safar, mind taking a peek?

File: src/coreclr/tools/aot/Mono.Linker.Tests/Mono.Linker.Tests.csproj [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>
     <Nullable>enable</Nullable>
+    <NoWarn>$(NoWarn);SYSLIB1045</NoWarn>
```
Comment (marek-safar) [inline src/coreclr/tools/aot/Mono.Linker.Tests/Mono.Linker.Tests.csproj:6]: Why is it not picking up test rules suppressions?

File: src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/InternalCalls.cs [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 using Internal.Runtime;
 
+#pragma warning disable SYSLIB1054 // Use LibraryImport instead of DllImport
```
Comment (marek-safar) [inline src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/InternalCalls.cs:13]: consider adding reason why it's disabled

File: src/libraries/Common/src/System/Data/Common/DbConnectionOptions.Common.cs [hunk 3]
```diff
@@ -61,17 +47,43 @@ internal partial class DbConnectionOptions
...
+        [GeneratedRegex(ConnectionStringPatternOdbc, RegexOptions.ExplicitCapture)]
+        private static partial Regex CreateConnectionStringRegexOdbc();
+#else
+        private static Regex CreateConnectionStringRegex() => new Regex(ConnectionStringPattern, RegexOptions.ExplicitCapture | RegexOptions.Compiled);
+        private static Regex CreateConnectionStringRegexOdbc() => new Regex(ConnectionStringPatternOdbc, RegexOptions.ExplicitCapture | RegexOptions.Compiled);
+#endif
 #endif
         internal const string DataDirectory = "|datadirectory|";
 
 #pragma warning disable CA1823 // used in some compilations and not others
```
Comment (marek-safar) [inline src/libraries/Common/src/System/Data/Common/DbConnectionOptions.Common.cs:66]: Do we still need the suppression now when the fields are ifdef-ed?

**Review by marek-safar — APPROVED** [review]

File: src/coreclr/tools/aot/Mono.Linker.Tests/Mono.Linker.Tests.csproj (hunk 1 above)
Comment (stephentoub) [reply src/coreclr/tools/aot/Mono.Linker.Tests/Mono.Linker.Tests.csproj:6]: Switching to the tests config is currently configured under libraries:
https://github.com/dotnet/runtime/blob/d92e70f05b8de72758f3a990844e53c76b44d713/src/libraries/Directory.Build.props#L47-L50

**Pull request merged**